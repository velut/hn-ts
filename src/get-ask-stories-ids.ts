import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";

/**
 * `getAskStoriesIds` returns the list of up to 200 IDs of the "Ask HN" stories.
 *
 * @see {@link https://github.com/HackerNews/API#ask-show-and-job-stories}
 */
export async function getAskStoriesIds(): Promise<number[]> {
  const endpoint = `${hnApiBaseUrl}/askstories.json`;
  const res = await fetch(endpoint);
  const ids = await res.json();
  return ids;
}
