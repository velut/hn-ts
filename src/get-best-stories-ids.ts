import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";

/**
 * `getBestStoriesIds` returns the list of up to 500 IDs of the best HN stories.
 *
 * @see {@link https://github.com/HackerNews/API#new-top-and-best-stories}
 */
export async function getBestStoriesIds(): Promise<number[]> {
  const endpoint = `${hnApiBaseUrl}/beststories.json`;
  const res = await fetch(endpoint);
  const ids = await res.json();
  return ids;
}
