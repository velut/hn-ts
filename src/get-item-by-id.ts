import { getRawItemById } from "./get-raw-item-by-id";
import { Item } from "./item";
import { normalizeRawItem } from "./normalize-raw-item";

/**
 * `getItemById` returns the `Item` with the given ID.
 * If the item is not available, `getItemById` returns `undefined`.
 *
 * @see {@link Item}
 * @see {@link https://github.com/HackerNews/API#items}
 */
export async function getItemById({
  id,
}: {
  id: number;
}): Promise<Item | undefined> {
  const rawItem = await getRawItemById({ id });
  if (!rawItem) {
    return undefined;
  }

  return normalizeRawItem({ rawItem });
}
