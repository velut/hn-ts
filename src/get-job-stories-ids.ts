import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";

/**
 * `getJobStoriesIds` returns the list of up to 200 IDs of the job HN stories.
 *
 * @see {@link https://github.com/HackerNews/API#ask-show-and-job-stories}
 */
export async function getJobStoriesIds(): Promise<number[]> {
  const endpoint = `${hnApiBaseUrl}/jobstories.json`;
  const res = await fetch(endpoint);
  const ids = await res.json();
  return ids;
}
