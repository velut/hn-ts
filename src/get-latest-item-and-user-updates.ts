import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";
import { Updated } from "./updated";

/**
 * `getLatestItemAndUserUpdates` returns the latest updates for items and users.
 *
 * @see {@link https://github.com/HackerNews/API#changed-items-and-profiles}
 */
export async function getLatestItemAndUserUpdates(): Promise<Updated> {
  const endpoint = `${hnApiBaseUrl}/updates.json`;
  const res = await fetch(endpoint);
  const updates = await res.json();
  return updates;
}
