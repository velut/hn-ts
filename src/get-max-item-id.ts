import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";

/**
 * `getMaxItemId` returns the ID of the most recent item available from the API.
 *
 * @see {@link https://github.com/HackerNews/API#max-item-id}
 */
export async function getMaxItemId(): Promise<number> {
  const endpoint = `${hnApiBaseUrl}/maxitem.json`;
  const res = await fetch(endpoint);
  const id = await res.json();
  return id;
}
