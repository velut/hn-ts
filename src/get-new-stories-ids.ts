import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";

/**
 * `getNewStoriesIds` returns the list of up to 500 IDs of the newer HN stories.
 *
 * @see {@link https://github.com/HackerNews/API#new-top-and-best-stories}
 */
export async function getNewStoriesIds(): Promise<number[]> {
  const endpoint = `${hnApiBaseUrl}/newstories.json`;
  const res = await fetch(endpoint);
  const ids = await res.json();
  return ids;
}
