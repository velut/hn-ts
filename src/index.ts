/**
 * This package TODO:
 *
 * @packageDocumentation
 */

export * from "./get-ask-stories-ids";
export * from "./get-best-stories-ids";
export * from "./get-item-by-id";
export * from "./get-job-stories-ids";
export * from "./get-max-item-id";
export * from "./get-new-stories-ids";
export * from "./get-raw-item-by-id";
export * from "./get-raw-user-by-id";
export * from "./get-show-stories-ids";
export * from "./get-top-stories-ids";
export * from "./get-user-by-id";
export * from "./hn-api-base-url";
export * from "./item";
export * from "./item-type";
export * from "./raw-item";
export * from "./raw-item-type";
export * from "./raw-user";
export * from "./user";
