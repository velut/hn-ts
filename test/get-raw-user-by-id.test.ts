/** @jest-environment setup-polly-jest/jest-environment-node */
// See https://netflix.github.io/pollyjs/#/test-frameworks/jest-jasmine?id=supported-test-runners

import NodeHttpAdapter from "@pollyjs/adapter-node-http";
import FSPersister from "@pollyjs/persister-fs";
import * as path from "path";
import { setupPolly } from "setup-polly-jest";
import { getRawUserById } from "../src";

describe("getRawUserById", () => {
  setupPolly({
    adapters: [NodeHttpAdapter],
    persister: FSPersister,
    persisterOptions: {
      fs: {
        recordingsDir: path.resolve(__dirname, "./__recordings__"),
      },
    },
    recordFailedRequests: true,
  });

  it("returns an existing user", async () => {
    expect.assertions(3);

    const user = await getRawUserById({ id: "velut" });

    expect(user).toBeDefined();
    expect(user!.id).toEqual("velut");
    expect(user).toMatchSnapshot();
  });

  it("returns undefined if a user with the given ID is not found", async () => {
    expect.assertions(1);

    const user = await getRawUserById({ id: "velut__________does_not_exist" });

    expect(user).toBeUndefined();
  });
});
