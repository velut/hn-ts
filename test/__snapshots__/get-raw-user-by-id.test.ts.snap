// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRawUserById returns an existing user 1`] = `
{
  "about": "GitHub: https:&#x2F;&#x2F;github.com&#x2F;velut&#x2F;",
  "created": 1556648526,
  "id": "velut",
  "karma": 9,
  "submitted": [
    31704327,
    29696636,
    27712220,
    27121843,
    27112535,
    27112241,
    27107840,
    27107832,
    25497292,
    25496976,
  ],
}
`;
